packages/data-context/src/codegen/spec-options.ts

```diff
@@ -18,18 +21,104 @@ interface CodeGenOptions {
 //   Button.foo.js  -> Button.foo-copy-1.js
 export const expectedSpecExtensions = ['.cy', '.spec', '.test', '-spec', '-test', '_spec']
 
+type ComponentExtension = '.cy.js' | '.cy.ts' | '.cy.jsx' | '.cy.tsx'
```

Comment:
Here is a neat trick:

```suggestion
type ComponentExtension = `cy.${'js' | 'ts' | 'jsx' | 'tsx'}`
```